safe_control_gym/experiments/base_experiment.py

```diff
@@ -84,48 +95,61 @@ def _execute_evaluations(self, n_episodes=None, n_steps=None, log_freq=None):
             raise ValueError('One of n_episodes or n_steps must be defined.')
         elif n_episodes is not None and n_steps is not None:
             raise ValueError('Only one of n_episodes or n_steps can be defined.')
+        if ctrl is None:
+            ctrl = self.ctrl
+        if env is None:
+            env = self.env
 
         # initialize
-        sim_steps = log_freq // self.env.CTRL_FREQ if log_freq else 1
+        sim_steps = log_freq // env.CTRL_FREQ if log_freq else 1
         steps, trajs = 0, 0
-        obs, info = self._evaluation_reset(ctrl_data=None, sf_data=None)
+        obs, info = self._evaluation_reset(ctrl=ctrl,
+                                           env=env,
+                                           ctrl_data=None,
+                                           sf_data=None)
         ctrl_data = defaultdict(list)
         sf_data = defaultdict(list)
 
         if n_episodes is not None:
             while trajs < n_episodes:
-                action = self._select_action(obs=obs, info=info)
+                action = ctrl.select_action(obs=obs, info=info)
```

Comment:
What is the reason/effect of the change from `_select_action()` to `select_action()`?

Reply:
This change is replacing the experiment _select_action with the controllers select action, which would remove the safety filter completely from the experiment class

Reply:
ok

Reply:
So (to be clear) it should be removed, this is typo

Reply:
good catch. Updated this and also modified `_select_action()` to take in a `ctrl` and `env` and defaults to `False` is `None` are provided to maintain performance.